src/components/sdk/MeetingControls/AudioInputVFControl.tsx

```diff
@@ -110,6 +113,9 @@ const AudioInputVFControl: React.FC<Props> = ({
             setIsLoading(true);
             const receivedDevice = audioInputSelectionToDevice(device.deviceId);
             const currentDevice = await addVoiceFocus(receivedDevice);
+
+            await (currentDevice as VoiceFocusTransformDevice).observeMeetingAudio(audioVideo as AudioVideoFacade);
```

Comment:
From the logic I see, you want to call `observeMeetingAudio` on the selected `currentDevice` if it is `VoiceFocusTransformDevice`.

This component code currently is already subscribing to change in audio input transform device. so the moment it changes the `device` state is also updated. In React, its a good practice to separate the concern into its own hooks. [Reference](https://reactjs.org/docs/hooks-effect.html#tip-use-multiple-effects-to-separate-concerns).

Based on this, I think it would be more cleaner to understand below code and you wont have to duplicate it in `onVFCheckboxChange` and here.

```tsx
useEffect(() => {
  if (!audioVideo) {
    return;
  }
  // This ensures that the selected current device is an Amazon Voice Focus transform device.
  if (isVoiceFocusEnabled) {
    await (currentDevice as VoiceFocusTransformDevice).observeMeetingAudio(audioVideo as AudioVideoFacade);
  }
}, [audioVideo, isVoiceFocusEnabled, device]);
```

This hook can go after the third `useEffect` in the existing code I think since in React the order of the hooks also matters.

Request you to update this to implement this using an extra `useEffect` hook. Tagging @jing-chen1 more help on this change.

Reply:
Is there a way to `unobserveMeetingAudio`? Suggesting removing the observer once the selected device is not an VoiceFocusTransformDevice anymore. Else, multiple observers are going to get added here.

Reply:
Observer is [Set](https://github.com/aws/amazon-chime-sdk-js/blob/147e7bb8f0b38e13410a50956c921d54cd872f44/src/audiomixcontroller/DefaultAudioMixController.ts#L22), which has dedup logic itself. same for [addFarEndStream](https://github.com/aws/amazon-chime-sdk-js/blob/147e7bb8f0b38e13410a50956c921d54cd872f44/src/voicefocus/VoiceFocusTransformDevice.ts#L259).

Reply:
Got it thanks for the explanation! Then the multiple observers wont be an issue, but I posted a general comment on how we plan to disable it? Will the feature support enabling-disabling ER when VF is enabled-disabled?